src/protocols/secure_channel/CASEServer.h

```diff
@@ -44,6 +44,7 @@ class CASEServer : public SessionEstablishmentDelegate,
         if (mExchangeManager != nullptr)
         {
             mExchangeManager->UnregisterUnsolicitedMessageHandlerForType(Protocols::SecureChannel::MsgType::CASE_Sigma1);
+            mExchangeManager = nullptr;
```

Comment:
@bzbarsky-apple @mrjerryjohns 

Made this change to fix ordering issue:
  - if CASEServer has a longer lifetime than the exchange manager associated with it (like the tests), then both Shutdown is called "normally" and in the destructor
  - by the time the destructor is called, the exchange manager may already be invalid

Reply:
CASEServer must not outlive the exchange manager.  Why is that happening in tests?  Can we fix them to not do that?

The shutdown change is ok, I think, but the lifetime ordering still needs to be right.

Reply:
I can fix the tests, but generally the logic was likely not ok: it relied on static object lifetime before that happened to work.

I am trying to not have PRs spiral too much into yak shaving.

Reply:
Fwiw, the logic was generally sane-ish fine: CASEServer is inited with ExchangeManager, so you have to tear down CASEServer before tearing down EXchangeManager, and the tests were just not doing that....

Reply:
@andy31415 Yes, I have no problem with this change.  I just think we should _also_ fix the test to do what the SDK itself does.